zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceMigrationPreconditionChecker.java

```diff
@@ -95,8 +96,8 @@ public final class ProcessInstanceMigrationPreconditionChecker {
   private static final String ERROR_ACTIVE_ELEMENT_WITH_BOUNDARY_EVENT =
       """
               Expected to migrate process instance '%s' \
-              but active element with id '%s' has a boundary event. \
-              Migrating active elements with boundary events is not possible yet.""";
+              but active element with id '%s' has one or more boundary events of types '%s'. \
```

Comment:
❓It is quite unlikely but there is still a possibility that users can rely on this error message, right? I do not think they are acting based on the message but this change can still break backward compatibility. WDYT?

Reply:
You're right, it is breaking behavior, but almost all changes do. 

I'd look at this change differently. They are two different errors. Users could have built around the old error and still can. It's just no longer in use. The changes produce a new error that users can build around.

We have [a clear definition](https://docs.camunda.io/docs/next/apis-tools/zeebe-api/gateway-service/#errors-9) of the possible errors. The exact error messages are not part of this. This error is part of the only error that users should expect changes to:
>A mapping instruction refers to element in unsupported scenarios. (i.e. migrating active elements with event subscriptions will be supported later on)

It's also documented as such in our [limitations](https://docs.camunda.io/docs/next/components/concepts/process-instance-migration/#limitations).

So, IMO users must be fine with us making changes here. What do you think?

Reply:
The documentation is a good point. The change can still break behaviour but I think, it is just a bad design for the API consumer side. 👍